### ISSUE TITLE
64 bit annotations only shown with 53 bit precision in UI

### ISSUE THREAD

**Issue by nollbit** [openzipkin/zipkin#1372]
We have a few binary annotations in our spans (emitted by Finagle using `Trace.binaryAnnotation`)

```
      {
        "key": "XXXXXXId",
        "value": 793024884742246336,
        "type": "I64",
        "endpoint": {
          "serviceName": "XXXXXXX",
          "ipv4": "10.19.XXX.XXX",
          "port": 10579
        }
      },
```

In the UI, however, the value shows as `793024884742246400`, due to the 53 bit javascript limitation. Perhaps `I64` should be sent as strings from the back-end?

**Comment by eirslett** [comment]
Yes, we should send them as strings!

**Comment by codefromthecrypt** [comment]
yeah please don't use types besides string (default). They are only present
as it was in the original thrift definition. However they are not
supported. When logging binary annotations for key lookup etc, stringify
them

**Comment by eirslett** [comment]
I suspect this problem is on the zipkin side. The JSON that comes from the Java server could be bad. If it sends the IDs as numbers instead of strings? Or are they sent as strings from the Zipkin server?

**Comment by nollbit** [comment]
Numbers.

The JSON above is copied from the response from `/api/v1/trace`.

**Comment by codefromthecrypt** [comment]
> I suspect this problem is on the zipkin side.
> 
> depends on the level of abstraction we define for this :) we've not
> "supported" anything besides string for user-defined binary annotations and
> almost certainly future formats won't support it. When people file issues
> like this we should probably let them know that this is not a feature that
> works (ex search won't work etc).
> 
> The JSON that comes from the Java server could be bad. If it sends the IDs
> as numbers instead of strings? Or are they sent as strings from the Zipkin
> server?
> 
> Ids are sent as strings, but this is a binary annotation. We could define a
> string encoding for when people use the unsupported I64 type, just I'm not
> sure how productive it is to invest in that.

**Comment by codefromthecrypt** [comment]
I guess it is probably this..

  def recordBinary(key: String, value: Any): Unit = {
    record(Annotation.BinaryAnnotation(key, value))
  }

Meh.. ok :) I'll raise a PR since yeah only thing we could do to help
is change zipkin-finagle to re-type the thing.

**Comment by codefromthecrypt** [comment]
https://github.com/openzipkin/zipkin/pull/1373

**Comment by codefromthecrypt** [comment]
fix going out now in 1.14.2
